uptane/clients/primary.py

```diff
@@ -1154,13 +1159,22 @@ def validate_time_attestation(self, timeserver_attestation):
     # Extract actual time from the timeserver's signed attestation.
     new_timeserver_time = timeserver_attestation['signed']['time']
 
+    # Make sure the format is understandable to us before saving the
+    # attestation and time.  Convert to a UNIX timestamp.
+    new_timeserver_time_unix = int(tuf.formats.datetime_to_unix_timestamp(
+        iso8601.parse_date(new_timeserver_time)))
+    tuf.formats.UNIX_TIMESTAMP_SCHEMA.check_match(new_timeserver_time_unix)
+
     # Save validated time.
     self.all_valid_timeserver_times.append(new_timeserver_time)
 
     # Save the attestation itself as well, to provide to Secondaries (who need
     # not trust us).
     self.all_valid_timeserver_attestations.append(timeserver_attestation)
 
+    # Set the client's clock.  This will be used instead of system time by TUF.
+    tuf.conf.CLOCK_OVERRIDE = new_timeserver_time_unix
+
```

Comment:
This seems like a serious side effect. Should it be mentioned in the function's docstring?

Reply:
I think that would be best, yes; thanks.  This is the current design in Uptane, though, a ratcheting "clock" based on verified timeserver attestations, so it should be unsurprising to users.

Reply:
While it looks like the docstring already says "the Primary's time is updated and the attestation will be saved so that it can be provided to Secondaries.", I'm going to change the function name and write a better docstring.